test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,272 @@ package scyllacluster
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		if len(nodes.Items) != 1 {
+			g.Skip(
+				"This test supports running at most 1 Node, as it relies on the fact that all Pods land on the same Node." +
+					"Replace with actual Node removal logic when CI env is bigger",
+			)
+		}
+
+		testStorageClassName := f.Namespace()
+
 		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
-		sc.Spec.Datacenter.Racks[0].Members = 3
 		sc.Spec.AutomaticOrphanedNodeCleanup = true
+		sc.Spec.Datacenter.Racks[0].Members = 3
+		sc.Spec.Datacenter.Racks[0].Storage.StorageClassName = pointer.Ptr(testStorageClassName)
+
+		// This test to trigger the orphaned PV cleanup is updating NodeAffinity of PV used by ScyllaCluster to not exiting Node.
+		// The problem is, that local volume storage provisioners which we use in CI, may set this field, which is immutable.
+		// To overcome this, the ScyllaCluster is using custom StorageClass not backed by any provisioner running in the cluster.
+		// ScyllaCluster is going to request a PVC from that StorageClass, and the test is going to request a clone of the original PVC
+		// from the default StorageClass to get any storage. Then the bound PV is rebounded to the original PVC
+		// but with empty NodeAffinity. This allows the test to trigger the orphaned PV cleanup logic.
+		go func() {
+			defer g.GinkgoRecover()
+
+			lw := &cache.ListWatch{
+				ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).List(ctx, options)
+				}),
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Watch(ctx, options)
+				},
+			}
+
+			_, err := watchtools.UntilWithSync(ctx, lw, &corev1.PersistentVolumeClaim{}, nil, func(e watch.Event) (bool, error) {
+				switch t := e.Type; t {
+				case watch.Added:
+					pvc := e.Object.(*corev1.PersistentVolumeClaim)
+					if pvc.Spec.StorageClassName == nil || *pvc.Spec.StorageClassName != testStorageClassName {
+						return false, nil
+					}
+
+					clone := &corev1.PersistentVolumeClaim{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: fmt.Sprintf("clone-%s", pvc.Name),
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes:      pvc.Spec.AccessModes,
+							Selector:         pvc.Spec.Selector,
+							Resources:        pvc.Spec.Resources,
+							VolumeName:       "",
+							StorageClassName: nil,
+							VolumeMode:       pvc.Spec.VolumeMode,
+							DataSource:       pvc.Spec.DataSource,
+							DataSourceRef:    pvc.Spec.DataSourceRef,
+						},
+					}
+
+					framework.Infof("Creating clone PVC for %q", pvc.Name)
+					clone, err = f.KubeClient().CoreV1().PersistentVolumeClaims(f.Namespace()).Create(ctx, clone, metav1.CreateOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					framework.Infof("Creating Pod a clone PVC consumer")
```

Comment:
can you remind me why we don't just create a PV instead of a pod with dummy pvc?

Reply:
I need storage provisioned by default storage class, and our local static provisioner is WaitForFirstConsumer.

Reply:
i think you should be able to create the PV on it's behalf with hostPath storage and fake it but you aim is likely to be sure about compatibility / testing coverage which is a good goal

Reply:
> I need storage provisioned by default storage class

Do you need it to use the default storage class, or just to use the local static provisioner? Because if it's the latter you could probably just create a corresponding storage class with the Immediate binding mode and create the pvc without consuming it.

Reply:
> Do you need it to use the default storage class, or just to use the local static provisioner?

I need storage in order to allow Scylla to boot up.

>   create a corresponding storage class with the Immediate binding mode and create the pvc without consuming it.

Hack on top of another hack.

It would create a volume with Local volume type which cannot have empty NodeAffinity. I would have to clone it anyway to rewrite it into hostPath volume type without NodeAffinity